utils/swift_build_support/swift_build_support/host_specific_configuration.py

```diff
@@ -203,6 +205,13 @@ def __init__(self, host_target, args):
                         "check-swift{}-optimize_none_with_implicit_dynamic-{}"
                         .format(subset_suffix, name))
 
+            # Only pull in these flags when cross-compiling with
+            # --cross-compile-hosts.
+            if deployment_target_name != args.host_target and \
+                host_target != args.host_target:
+                self.swift_flags = deployment_target.swift_flags(args)
```

Comment:
I really dislike how this mutates global state in self outside of self. I would feel more comfortable if there was an explicit method on self that takes in the deployment_target and sets flags. Something like that.

Reply:
Not sure what you mean, as `self` in this case is `HostSpecificConfiguration` for which these lines I'm adding are in the initializer for that class. This is in keeping with how this initializer configures other variables like `self.sdks_to_configure` above.

Reply:
I meant outside of the class. It seems like one could make a separate method that does this on self that way it is self contained and given a nice comment explaining that we are changing the deployment target or updating the deployment target.